lib/ansible/module_utils/net_tools/nios/api.py

```diff
@@ -356,18 +356,26 @@ def run(self, ib_obj_type, ib_spec):
                                 res = ref
                 if (ib_obj_type in (NIOS_A_RECORD, NIOS_AAAA_RECORD, NIOS_PTR_RECORD, NIOS_SRV_RECORD)):
                     # popping 'view' key as update of 'view' is not supported with respect to a:record/aaaa:record/srv:record/ptr:record
-                    if 'ipv4addrs' in proposed_object:
-                        if 'add' in proposed_object['ipv4addrs'][0]:
-                            run_update, proposed_object = self.check_if_add_remove_ip_arg_exists(proposed_object)
-                            if run_update:
-                                res = self.update_object(ref, proposed_object)
-                                result['changed'] = True
-                            else:
-                                res = ref
-                if (ib_obj_type in (NIOS_A_RECORD, NIOS_AAAA_RECORD)):
-                    # popping 'view' key as update of 'view' is not supported with respect to a:record/aaaa:record
                     proposed_object = self.on_update(proposed_object, ib_spec)
                     del proposed_object['view']
+                    if (ib_obj_type == NIOS_A_RECORD):
+                        # resolves issue where multiple a_records with same name and different IP address
+                        try:
+                            ipaddr_obj = self.module._check_type_dict(proposed_object['ipv4addr'])
+                            ipaddr = ipaddr_obj['new_ipv4addr']
```

Comment:
#64045 implementation where `nios_a_record` new params `new_ipv4addr` and `old_ipv4addr` are introduced need to be moved to separate and new PR, else this PR cannot be backported to Ansible 2.9.

Reply:
@krisvasudevan have u opened a new PR to handle `nios_a_record` scenario.

Reply:
Have opened a PR for both nios_a_record and nios_txt_record scenario